src/util/task.rs

```diff
@@ -17,18 +17,20 @@ pub async fn join_all(tasks: Vec<Task>) -> AppResult {
     tokio::spawn(async {
       if let Err(e) = task.await {
         if let Some(sender) = sender {
-          sender
-            .send(e)
-            .await
-            .unwrap_or_else(|_| unreachable!("This channel never closed."));
+          let _ = sender.send(e).await;
```

Comment:
I think this change does not explain why to ignore the result, but the previous code does. It might be better to add a comment : "This channel never closed."